InternalPythonModules/android/operabrowser.py

```diff
@@ -0,0 +1,200 @@
+                    cookiesResultSet = cookiesDb.runQuery("SELECT host_key, name, value, creation_utc FROM cookies")
+                    if cookiesResultSet is not None:
+                        while cookiesResultSet.next():
+                            createTime = cookiesResultSet.getLong("creation_utc") / 1000000 - 11644473600 # Webkit time
+                            cookiesDbHelper.addWebCookie( cookiesResultSet.getString("host_key"),
+                                                        createTime,  
+                                                        cookiesResultSet.getString("name"),
+                                                        cookiesResultSet.getString("value"),
+                                                        self.progName)
+
+                except SQLException as ex:
+                    self._logger.log(Level.WARNING, "Error processing query results for Opera cookies.", ex)
+                except (TskCoreException, BlackboardException) as ex:
+                    self._logger.log(Level.WARNING, "Failed to add Opera cookie artifacts.", ex)
+                finally:      
+                    cookiesDb.close()                    
+            	
+        
+
+    def analyzeHistory(self, dataSource, fileManager, context):
+            historyDbs = AppSQLiteDB.findAppDatabases(dataSource, "History", True, "com.opera.browser")
+            for historyDb in historyDbs:
+                try:
+                    historyDbHelper = WebBrowserArtifactsHelper(Case.getCurrentCase().getSleuthkitCase(),
+                                            self.moduleName, historyDb.getDBFile())
+                    historyResultSet = historyDb.runQuery("SELECT url, title, last_visit_time FROM urls")
+                    if historyResultSet is not None:
+                        while historyResultSet.next():
+                            accessTime = historyResultSet.getLong("last_visit_time") / 1000000 - 11644473600
+                            historyDbHelper.addWebHistory( historyResultSet.getString("url"),
+                                                        accessTime,
+                                                        "",     # referrer
+                                                        historyResultSet.getString("title"),
+                                                        self.progName)
+                except SQLException as ex:
+                    self._logger.log(Level.WARNING, "Error processing query results for Opera history.", ex)
+                except (TskCoreException, BlackboardException) as ex:
+                    self._logger.log(Level.WARNING, "Failed to add Opera history artifacts.", ex)
+                finally:        
+                    historyDb.close()                    
+                
+        
+
+    def analyzeDownloads(self, dataSource, fileManager, context):
+            downloadsDbs = AppSQLiteDB.findAppDatabases(dataSource, "History", True, "com.opera.browser")
```

Comment:
Same comments from above and the installedprograms hold for all the rest of these.